rados/watcher.go

```diff
@@ -0,0 +1,385 @@
+//go:build ceph_preview
+// +build ceph_preview
+
+package rados
+
+/*
+#cgo LDFLAGS: -lrados
+#include <stdlib.h>
+#include <rados/librados.h>
+extern void watchNotifyCb(void*, uint64_t, uint64_t, uint64_t, void*, size_t);
+extern void watchErrorCb(void*, uint64_t, int);
+*/
+import "C"
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"time"
+	"unsafe"
+)
+
+type (
+	WatcherId  uint64 // unique id of a Watcher.
+	NotifyId   uint64 // unique id of a NotifyEvent.
+	NotifierId uint64 // unique id of a notifying client.
+)
+
+// NotifyEvent is received by a watcher for each notification.
+type NotifyEvent struct {
+	Id NotifyId
+	WatcherId
+	NotifierId
```

Comment:
Maybe I'm being dense today, but wouldn't these make WatcherId and NotifierId embedded structs? Wouldn't it be clearer if they were actually fields?  I think go permits you to re-use the type name, eg:
```go
type NotifyEvent struct {
	Id NotifyId
	WatcherId WatcherId
	WatcherId NotifierId
	Data []byte
}
```

Also, I know in go-ceph we use "Id" a lot but the checker tools generally prefer "ID". Your choice, but I thought I'd mention it too.